cam6_4_122: Prescribed MAM5 volcanic forcings

fvitt opened the pull request:
This provides the ability to use prescribed MAM5 stratospheric aerosols.

Closes #1187

fvitt committed:
Simone's changes for fifth mode

	modified:   src/chemistry/utils/prescribed_strataero.F90

fvitt committed:
fix backwards compatibility issues

	modified:   src/chemistry/utils/prescribed_strataero.F90

fvitt committed:
add regression test

	new file:   cime_config/testdefs/testmods_dirs/cam/outfrq9s_cam7_presc_volc/shell_commands
	new file:   cime_config/testdefs/testmods_dirs/cam/outfrq9s_cam7_presc_volc/user_nl_cam
	new file:   cime_config/testdefs/testmods_dirs/cam/outfrq9s_cam7_presc_volc/user_nl_clm
	modified:   cime_config/testdefs/testlist_cam.xml

fvitt committed:
Use TEST_ONLY input file for reg test

	modified:   cime_config/testdefs/testmods_dirs/cam/outfrq9s_cam7_presc_volc/user_nl_cam

fvitt committed:
use preprocessor tag chem_proc5_0_07

cacraigucar commented on cime_config/testdefs/testmods_dirs/cam/outfrq9s_cam7_presc_volc/user_nl_cam:
```diff
@@ -0,0 +1,24 @@
+ 'N:VOLC_MMR1:/glade/campaign/cesm/cesmdata/inputdata/atm/cam/physprops/volc_camRRTMG_byradius_sigma1.6_mode1_c210211.nc',
+ 'N:VOLC_MMR2:/glade/campaign/cesm/cesmdata/inputdata/atm/cam/physprops/volc_camRRTMG_byradius_sigma1.6_mode2_c210211.nc',
+ 'N:VOLC_MMR3:/glade/campaign/cesm/cesmdata/inputdata/atm/cam/physprops/volc_camRRTMG_byradius_sigma1.2_mode3_c210211.nc',
+ 'N:VOLC_MMR5:/glade/campaign/cesm/cesmdata/inputdata/atm/cam/physprops/volc_camRRTMG_byradius_sigma1.2_mode3_c210211.nc'
```
These need to use a relative path

fvitt replied:
fixed

cacraigucar commented on src/chemistry/utils/prescribed_strataero.F90:
```diff
@@ -192,7 +202,7 @@ subroutine prescribed_strataero_register()
 
        ierr = pio_inq_varid( file_handle, 'so4mass_a1', varid )
-       three_mode = three_mode .and. (ierr.eq.PIO_NOERR)
+       three_mode = (ierr.eq.PIO_NOERR)
```
Since you are modifying this line, perhaps you could also use modern Fortran comparison syntax (i.e. ==) here and on line 218 and 220

fvitt replied:
fixed

fvitt committed:
fix issues raised by code reviewer

	modified:   cime_config/testdefs/testmods_dirs/cam/outfrq9s_cam7_presc_volc/user_nl_cam
	modified:   src/chemistry/utils/prescribed_strataero.F90

fvitt committed:
Merge remote-tracking branch 'ESCOMP/cam_development' into prescribed_mam5_strataero

fvitt committed:
Merge remote-tracking branch 'origin/chem_preproc_fix' into prescribed_mam5_strataero

fvitt committed:
update ChangeLog
